spatial/r3/box.go

```diff
@@ -0,0 +1,79 @@
+// Copyright ©2022 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package r3
+
+// Box is a 3D bounding box.
+type Box struct {
+	Min, Max Vec
+}
+
+// CenteredBox creates a Box with a given center and size.
+func CenteredBox(center, size Vec) Box {
+	half := Scale(0.5, size)
+	return Box{Min: Sub(center, half), Max: Add(center, half)}
+}
+
+// Size returns the size of the Box.
+func (a Box) Size() Vec {
+	return Sub(a.Max, a.Min)
+}
+
+// Center returns the center of the Box.
+func (a Box) Center() Vec {
+	return Add(a.Min, Scale(0.5, a.Size()))
+}
+
+// Vertices returns a slice of the 8 vertices
+// corresponding to each of the Box's corners.
+//
+// Vertex ordering between 0-3 and 4-7 outline
+// 2D boxes in the XY plane where vertices 0-3 are in Z=Box.Min.Z.
+// One can then construct the edges with indices for the
+// return value of Vertices:
+//  edges := [12][2]int{
+//   {0, 1}, {1, 2}, {2, 3}, {3, 0},
+//   {4, 5}, {5, 6}, {6, 7}, {7, 4},
+//   {0, 4}, {1, 5}, {2, 6}, {3, 7},
+//  }
+func (a Box) Vertices() []Vec {
+	return []Vec{
+		a.Min,                                // 0
+		{X: a.Max.X, Y: a.Min.Y, Z: a.Min.Z}, // 1
+		{X: a.Max.X, Y: a.Max.Y, Z: a.Min.Z}, // 2
+		{X: a.Min.X, Y: a.Max.Y, Z: a.Min.Z}, // 3
+		{X: a.Min.X, Y: a.Min.Y, Z: a.Max.Z}, // 4
+		{X: a.Max.X, Y: a.Min.Y, Z: a.Max.Z}, // 5
+		a.Max,                                // 6
+		{X: a.Min.X, Y: a.Max.Y, Z: a.Max.Z}, // 7
+	}
+}
```

Comment:
Never put in comments what can go in code.

```suggestion
	return []Vec{
		0: a.Min,
		1: {X: a.Max.X, Y: a.Min.Y, Z: a.Min.Z},
		2: {X: a.Max.X, Y: a.Max.Y, Z: a.Min.Z},
		3: {X: a.Min.X, Y: a.Max.Y, Z: a.Min.Z},
		4: {X: a.Min.X, Y: a.Min.Y, Z: a.Max.Z},
		5: {X: a.Max.X, Y: a.Min.Y, Z: a.Max.Z},
		6: a.Max,
		7: {X: a.Min.X, Y: a.Max.Y, Z: a.Max.Z},
	}
}
```